src/observers/BaseStatusBarItemObserver.ts

```diff
@@ -11,17 +11,19 @@ export abstract class BaseStatusBarItemObserver {
     constructor(private statusBarItem: StatusBarItem) {
     }
 
-    public SetAndShowStatusBar(text: string, command: string, color?: string) {
+    public SetAndShowStatusBar(text: string, command: string, color: string, tooltip: string) {
```

Comment:
Consider making tooltip an optional parameter? That way the callsites that pass empty string for tooltip can just pass nothing instead? Marginally more readable, I guess.

Reply:
I would probably do the same for "color", since you seem to pass green or nothing.